task_mapper/heft_mulhome/original/heft_dup.py

```diff
@@ -118,7 +118,9 @@ def cal_avg_comm(self, task1, task2):
             for j in range(self.num_processor):
                 if i==j: continue
                 res += self.cal_comm_quadratic(self.data[task1.number][task2.number],self.quaratic_profile[i][j])
-
+        if(res <= 0):
+            print("Aquired negative communication cost, please redeploy DRUPE, HEFT will terminate !!!")
```

Comment:
can you use a logging.error() call for printing instead?

is there any cleanup you can add to the exit routine?

Reply:
Can you give me an example of what to include in the clean up? I think usually systems would dump something on disk etc in clean up, but we don't need it here

Reply:
Would it make sense to execute the destroy scripts if this fails instead of just printing to do so?